### ISSUE TITLE
Why is my COCO result so bad?

### ISSUE THREAD

**Issue by vvision0** [ultralytics/yolov5#6733]
### Search before asking

- [X] I have searched the YOLOv5 [issues](https://github.com/ultralytics/yolov5/issues) and [discussions](https://github.com/ultralytics/yolov5/discussions) and found no similar questions.

### Question

I use the command:
**python val.py --weights yolov5s.pt --data coco.yaml --img 640**
and the local terminal shows:
Class     Images     Labels          P          R     mAP@.5 mAP@.5:.95: 100%|██████████| 157/157 [00:50<00:00,  3.12it/s]
 all       5000      36335      0.678      0.502      0.557      0.361

Average Precision  (AP) @[ IoU=0.50      | area=   all | maxDets=100 ] = 0.003
 Average Precision  (AP) @[ IoU=0.75      | area=   all | maxDets=100 ] = 0.002
 Average Precision  (AP) @[ IoU=0.50:0.95 | area= small | maxDets=100 ] = 0.002
 Average Precision  (AP) @[ IoU=0.50:0.95 | area=medium | maxDets=100 ] = 0.002
 Average Precision  (AP) @[ IoU=0.50:0.95 | area= large | maxDets=100 ] = 0.004
 Average Recall     (AR) @[ IoU=0.50:0.95 | area=   all | maxDets=  1 ] = 0.047
 Average Recall     (AR) @[ IoU=0.50:0.95 | area=   all | maxDets= 10 ] = 0.093
 Average Recall     (AR) @[ IoU=0.50:0.95 | area=   all | maxDets=100 ] = 0.100
 Average Recall     (AR) @[ IoU=0.50:0.95 | area= small | maxDets=100 ] = 0.066
 Average Recall     (AR) @[ IoU=0.50:0.95 | area=medium | maxDets=100 ] = 0.117

Pycocotools version is 2.0.0.2,  it was installed by the command:
**pip install pycocotools-windows**
because my computer system is windows 11 and the command:
**pip install pycocotools**
is not work.

Does the version of pycocotools result in this outcome? How can i fix it?

**Comment by glenn-jocher** [comment]
@vvision0 it appears you may have environment problems. Please ensure you meet all dependency requirements if you are attempting to run YOLOv5 locally.  If in doubt, create a new virtual [Python 3.9](https://www.python.org/) environment, clone the latest repo ([code changes daily](https://github.com/ultralytics/yolov5/commits/master)), and `pip install` [requirements.txt](https://github.com/ultralytics/yolov5/blob/master/requirements.txt) again from scratch.

💡 ProTip! Try one of our verified environments below if you are having trouble with your local environment.


## Requirements

[**Python>=3.7.0**](https://www.python.org/) with all [requirements.txt](https://github.com/ultralytics/yolov5/blob/master/requirements.txt) installed including [**PyTorch>=1.7**](https://pytorch.org/get-started/locally/). To get started:
```bash
git clone https://github.com/ultralytics/yolov5  # clone
cd yolov5
pip install -r requirements.txt  # install
```

[Models](https://github.com/ultralytics/yolov5/tree/master/models) and [datasets](https://github.com/ultralytics/yolov5/tree/master/data) download automatically from the latest YOLOv5 [release](https://github.com/ultralytics/yolov5/releases) when first requested.

## Environments

YOLOv5 may be run in any of the following up-to-date verified environments (with all dependencies including [CUDA](https://developer.nvidia.com/cuda)/[CUDNN](https://developer.nvidia.com/cudnn), [Python](https://www.python.org/) and [PyTorch](https://pytorch.org/) preinstalled):

- **Google Colab and Kaggle** notebooks with free GPU: <a href="https://colab.research.google.com/github/ultralytics/yolov5/blob/master/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolov5"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **Google Cloud** Deep Learning VM. See [GCP Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial/)
- **Amazon** Deep Learning AMI. See [AWS Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial/)
- **Docker Image**. See [Docker Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial/) <a href="https://hub.docker.com/r/ultralytics/yolov5"><img src="https://img.shields.io/docker/pulls/ultralytics/yolov5?logo=docker" alt="Docker Pulls"></a>


## Status

![CI CPU testing](https://github.com/ultralytics/yolov5/workflows/CI%20CPU%20testing/badge.svg)

If this badge is green, all [YOLOv5 GitHub Actions](https://github.com/ultralytics/yolov5/actions) Continuous Integration (CI) tests are currently passing. CI tests verify correct operation of YOLOv5 training ([train.py](https://github.com/ultralytics/yolov5/blob/master/train.py)), validation ([val.py](https://github.com/ultralytics/yolov5/blob/master/val.py)), inference ([detect.py](https://github.com/ultralytics/yolov5/blob/master/detect.py)) and export ([export.py](https://github.com/ultralytics/yolov5/blob/master/export.py)) on MacOS, Windows, and Ubuntu every 24 hours and on every commit.

**Comment by vvision0** [comment]
@glenn-jocher Thank you for your reply, I run the command:
**pip install -r requirements.txt**
and the terminal shows a lot of "already satisfied" with
**WARNING: Ignore distutils configs in setup.cfg due to encoding errors.**
in the last line.Does this warning have effect on the result?
I check the requirements one by one and everyone is satisfied except <wandb> and things in "Export" because i use pytorch ,and <"albumentations", "roboflow"> in "Extras".
I also tried the other's .json file such as results of Faster-Rcnn and the coco result shows correctly.
Then I debug the .json file in the exp folder and find that there are many items in one picture and few of them have high scores while many are very low. Will this have a effect?

**Comment by yuhao0613** [comment]
@vvision0 @glenn-jocher I also have the same problem.
when I run the command: `pip install -r requirements.txt`, the terminal shows a lot of `WARNING: Ignore distutils configs in setup.cfg due to encoding errors.` in the last line. I could not find any solutions on the Internet.

**Comment by EileenWang90** [comment]
@vvision0 @yuhao0613 I also have the same problem. 
I use the same command: `python val.py --weights yolov5s.pt --data coco.yaml --img 640`.  
And in coco.yaml, I modified the loction of coco dataset which influences the variable`is_coco` in val.py
> in val.py line158  **is_coco = isinstance(data.get('val'), str) and data['val'].endswith('coco/val2017.txt')  # COCO dataset**

After my modify, the variable`is_coco` bacame False, thus results in the following variable `class_map`. Then the classes for images was wrong.  
>in val.py line175  **class_map = coco80_to_coco91_class() if is_coco else list(range(1000))**

SO my solution is to make the variable`is_coco` bacome True, then the pycocotools is OK and the result is OK.

**Comment by vvision0** [comment]
I have solved this problem with a little trick.
Just train locally and a best.pt file will be generated, then create a google colab notebook and install the environment. Upload the best.pt file to the yolov5 folder in the colab and run:
**python val.py --weights best.pt --data coco.yaml --img 640**
then the results will be correct.

**Comment by diangu001** [comment]
my solution :
class_map = coco80_to_coco91_class()  # if is_coco else list(range(1000))

**Comment by imadgohar** [comment]
> ### Search before asking
> * [x]  I have searched the YOLOv5 [issues](https://github.com/ultralytics/yolov5/issues) and [discussions](https://github.com/ultralytics/yolov5/discussions) and found no similar questions.
> 
> ### Question
> I use the command: **python val.py --weights yolov5s.pt --data coco.yaml --img 640** and the local terminal shows: Class Images Labels P R mAP@.5 mAP@.5:.95: 100%|██████████| 157/157 [00:50<00:00, 3.12it/s] all 5000 36335 0.678 0.502 0.557 0.361
> 
> Average Precision (AP) @[ IoU=0.50 | area= all | maxDets=100 ] = 0.003 Average Precision (AP) @[ IoU=0.75 | area= all | maxDets=100 ] = 0.002 Average Precision (AP) @[ IoU=0.50:0.95 | area= small | maxDets=100 ] = 0.002 Average Precision (AP) @[ IoU=0.50:0.95 | area=medium | maxDets=100 ] = 0.002 Average Precision (AP) @[ IoU=0.50:0.95 | area= large | maxDets=100 ] = 0.004 Average Recall (AR) @[ IoU=0.50:0.95 | area= all | maxDets= 1 ] = 0.047 Average Recall (AR) @[ IoU=0.50:0.95 | area= all | maxDets= 10 ] = 0.093 Average Recall (AR) @[ IoU=0.50:0.95 | area= all | maxDets=100 ] = 0.100 Average Recall (AR) @[ IoU=0.50:0.95 | area= small | maxDets=100 ] = 0.066 Average Recall (AR) @[ IoU=0.50:0.95 | area=medium | maxDets=100 ] = 0.117

How you come up with these results? I trained a model with .txt annotation files, Now I want to take Coco type results as you mentioned (My test set is in .JSON file) how I can get these results?

Thanks

**Comment by imadgohar** [comment]
@vvision0, I want to get area= small, medium and large results.

**Comment by glenn-jocher** [comment]
See Colab notebook for COCO validation example:
https://colab.research.google.com/github/ultralytics/yolov5/blob/master/tutorial.ipynb?hl=en

**Comment by imadgohar** [comment]
> See Colab notebook for COCO validation example: https://colab.research.google.com/github/ultralytics/yolov5/blob/master/tutorial.ipynb?hl=en

Thanks for your quick response, just want to confirm my model is trained on YOLO type annotation and on Val/test set I want Coco type results (where I can see small, medium and large) type results.
Is it possible, I only converted my test set annotations to .JSON format.

Thanks.

**Comment by imadgohar** [comment]
> @imadgohar no sorry, COCO validation is only intended for COCO dataset.

Thanks

**Comment by imadgohar** [comment]
@glenn-jocher  I trained the model with coco format data now I am getting this problem, what is the solution for it.
![yolo](https://user-images.githubusercontent.com/45845910/204725900-245f2703-dca8-4662-b2e6-dbdf7f2844e3.PNG)

**Comment by glenn-jocher** [comment]
@imadgohar this section of val is only intended for for COCO dataset.
